src/plugins/usage_collection/server/plugin.ts

```diff
@@ -13,81 +13,22 @@ import type {
 import { CollectorSet } from './collector';
-import type { Collector, CollectorOptions, UsageCollectorOptions } from './collector';
-import { setupRoutes } from './routes';
-
 import { UsageCountersService } from './usage_counters';
-import type { UsageCounter } from './usage_counters';
+import { registerUsageCountersSavedObjectTypes } from './usage_counters/saved_objects';
 
-/** Server's setup APIs exposed by the UsageCollection Service **/
-export interface UsageCollectionSetup {
-  /**
-   * Creates and registers a usage counter to collect daily aggregated plugin counter events
-   */
-  createUsageCounter: (type: string) => UsageCounter;
-  /**
-   * Returns a usage counter by type
-   */
-  getUsageCounterByDomainId: (type: string) => UsageCounter | undefined;
-  /**
-   * Creates a usage collector to collect plugin telemetry data.
-   * registerCollector must be called to connect the created collector with the service.
-   */
-  makeUsageCollector: <TFetchReturn, ExtraOptions extends object = {}>(
-    options: UsageCollectorOptions<TFetchReturn, ExtraOptions>
-  ) => Collector<TFetchReturn, ExtraOptions>;
-  /**
-   * Register a usage collector or a stats collector.
-   * Used to connect the created collector to telemetry.
-   */
-  registerCollector: <TFetchReturn, ExtraOptions extends object>(
-    collector: Collector<TFetchReturn, ExtraOptions>
-  ) => void;
-  /**
-   * Returns a usage collector by type
-   */
-  getCollectorByType: <TFetchReturn, ExtraOptions extends object>(
-    type: string
-  ) => Collector<TFetchReturn, ExtraOptions> | undefined;
-  /**
-   * Fetches the collection from all the registered collectors
-   * @internal: telemetry use
-   */
-  bulkFetch: <TFetchReturn, ExtraOptions extends object>(
-    esClient: ElasticsearchClient,
-    soClient: SavedObjectsClientContract,
-    collectors?: Map<string, Collector<TFetchReturn, ExtraOptions>>
-  ) => Promise<Array<{ type: string; result: unknown }>>;
-  /**
-   * Converts an array of fetched stats results into key/object
-   * @internal: telemetry use
-   */
-  toObject: <Result extends Record<string, unknown>, T = unknown>(
-    statsData?: Array<{ type: string; result: T }>
-  ) => Result;
-  /**
-   * Rename fields to use API conventions
-   * @internal: monitoring use
-   */
-  toApiFieldNames: (
-    apiData: Record<string, unknown> | unknown[]
-  ) => Record<string, unknown> | unknown[];
-  /**
-   * Creates a stats collector to collect plugin telemetry data.
-   * registerCollector must be called to connect the created collector with the service.
-   * @internal: telemetry and monitoring use
-   */
-  makeStatsCollector: <TFetchReturn, ExtraOptions extends object = {}>(
-    options: CollectorOptions<TFetchReturn, ExtraOptions>
-  ) => Collector<TFetchReturn, ExtraOptions>;
-}
+/** Plugin's setup API **/
+export type UsageCollectionSetup = ICollectorSet & UsageCountersServiceSetup;
```

Comment:
@afharo @Bamieh I have extracted all of the CollectorSet methods into a separate interface `ICollectorSet`, in an effort to make it cleaner / clearer that the `usage-collection` plugin offers 2 different things in the _setup_ contract.